### ISSUE TITLE
Unable to Reproduce the results for VinVL+VIVO on NoCaps

### ISSUE THREAD

**Issue by joeyy5588** [microsoft/Oscar#131]
I tried to reproduce the results for VinVL+VIVO+SCST on NoCaps, but my result was off by a visible margin.

### Reported Results on NoCaps validation set
`"CIDEr": {"in-domain": 103.7, "near-domain": 95.6, "out-domain": 83.8, "entire": 94.3}, "SPICE": {"in-domain": 13.7, "near-domain": 13.4, "out-domain": 11.9, "entire": 13.1}}
`
### Reproduced results:
`"CIDEr": {"in-domain": 95.0, "near-domain": 91.1, "out-domain": 79.2, "entire": 89.3}, "SPICE": {"in-domain": 13.1, "near-domain": 12.9, "out-domain": 11.2, "entire": 12.6}}
`

I'll specify the data, scripts and codes I used during pretraining, cross_entropy opimization, CIDEr optimization and inference

#### Pretraining 
For OpenImages, I downloaded the feature files and label files released by this repo from the folloing links. [features](https://biglmdiag.blob.core.windows.net/vinvl/image_features/oi_X152C4_frcnnbig2_exp168model_0060000model.roi_heads.nm_filter_2_model.roi_heads.score_thresh_0.2/model_0060000/), [labels](https://biglmdiag.blob.core.windows.net/vinvl/pretrain_corpus/X152C4_frcnnbig2_exp168model_0060000model.roi_heads.nm_filter_2_model.roi_heads.score_thresh_0.2/oi)
#### Cross_entropy, CIDEr Optimization
I downloaded the coco_caption dataset following the instruction from [VinVL_DOWNLOAD.md](https://github.com/microsoft/Oscar/blob/master/VinVL_DOWNLOAD.md)
#### Inference
I downloaded the nocaps dataset following the instruction from [VinVL_DOWNLOAD.md](https://github.com/microsoft/Oscar/blob/master/VinVL_DOWNLOAD.md), which includes the feature files and label files for nocaps validation set.

#### Pretraining 
I slightly modify the [run_oscarplus_pretrain.py](https://github.com/microsoft/Oscar/blob/master/oscar/run_oscarplus_pretrain.py) to perform MLM on the tag sequence.
#### Cross_entropy, CIDEr Optimization, Inference
I leverage [run_captioning.py](https://github.com/microsoft/Oscar/blob/master/oscar/run_captioning.py) to perform cross_entropy optimization, CIDEr optimization on COCO, and evaluate on NoCaps validation set.

#### Pretraining
`batch_size=1024, lr=5e-05, maximum of 50 image regions and 15 tag tokens per image, trained for 160K iters, same masking strategy` Parameters are set according to the VIVO paper.
#### Cross_entropy
`batch_size=256, lr=5e-05, trained for 30 epochs` These parameters are set according to the VIVO paper, other parameters remain the default value in [run_captioning.py](https://github.com/microsoft/Oscar/blob/master/oscar/run_captioning.py).
#### CIDEr Optimization
`batch_size=14*8=112, lr=5e-6, trained for 40 epochs`
I found that the training scripts from [VinVL_DOWNLOAD.md](https://github.com/microsoft/Oscar/blob/master/VinVL_DOWNLOAD.md) can yield better performance than the setting in VIVO paper, so I used the scripts to perform CIDEr Optimization.
#### Inference
`max_gen_length=20, num_beams=5`

I'm wondering if I made mistakes during these steps. 
Really appreciate if you could clarify this, thanks!

**Comment by Jennifer-6** [comment]
can l have a contact number?

**Comment by joeyy5588** [comment]
You can reach me via email.

**Comment by Jennifer-6** [comment]
Is it possible to communicate in Chinese? If so, can you add me on WeChat?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/Oscar"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月17日(星期二) 晚上9:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/Oscar] Unable to Reproduce the results for VinVL+VIVO on NoCaps (#131)





 
You can reach me via email.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Jennifer-6** [comment]
in order to&nbsp; run run_captioning.py&nbsp; ,train.yaml is needed. train.yaml file is some required data(image feature,caption,labels),where is the train.yaml? or how to get train.yaml?




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/Oscar"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月17日(星期二) 晚上9:37
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/Oscar] Unable to Reproduce the results for VinVL+VIVO on NoCaps (#131)





 
You can reach me via email.
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by xiaoweihu** [comment]
Hi,

This repo is for VinVL pretraining. The VIVO pre-training loss is different from what is used in VinVL, e.g., the Hungarian matching. Hope this could help.

Best,
Xiaowei

**Comment by lhlclhl** [comment]
> Hi,
> 
> This repo is for VinVL pretraining. The VIVO pre-training loss is different from what is used in VinVL, e.g., the Hungarian matching. Hope this could help.
> 
> Best, Xiaowei

Hi, 
Is the VIVO pre-training publicly accessible? 
I cannot find any code or pre-trained models about VIVO.
